interfaces/builtin/utils.go

```diff
@@ -125,3 +126,58 @@ func verifySlotPathAttribute(slotRef *interfaces.SlotRef, attrs interfaces.Attre
 	}
 	return cleanPath, nil
 }
+
+// aareExclusivePatterns takes a string and generates deny alternations. Eg,
+// aareExclusivePatterns("foo") returns:
+// []string{
+//   "[^f]*",
+//   "f[^o]*",
+//   "fo[^o]*",
+// }
+func aareExclusivePatterns(orig string) []string {
+	// This function currently is only intended to be used with desktop
+	// prefixes as calculated by info.DesktopPrefix (the snap name and
+	// instance name, if present). To avoid having to worry about aare
+	// special characters, etc, perform ValidateDesktopPrefix() and return
+	// an empty list if invalid. If this function is modified for other
+	// input, aare/quoting/etc will have to be considered.
+	if !snap.ValidateDesktopPrefix(orig) {
+		return make([]string, 0)
```

Comment:
Could this return just `nil` here? Given that `nil` can be used as an empty slice?

Reply:
Done